pkg/sql/sqlbase/index_encoding.go

```diff
@@ -852,8 +854,11 @@ func EncodeInvertedIndexKeys(
 	} else {
 		val = tree.DNull
 	}
-
-	return EncodeInvertedIndexTableKeys(val, keyPrefix)
+	if !geoindex.IsEmptyConfig(&index.GeoConfig) {
+		return encodeGeoInvertedIndexTableKeys(val, keyPrefix, index)
```

Comment:
I'd prefer this switch to go inside of `EncodeInvertedIndexTableKeys`, because that function already has a switch between JSON and Array - so it feels more natural there. Won't you be able to add a case to `switch val.ResolvedType().Family()` rather than checking the index type as well, or am I missing something?

Reply:
I see, it's that span_builder is a different semantic use. Let's talk more in the meeting, I don't get this still.